yjit_codegen.c

```diff
@@ -183,6 +186,26 @@ yjit_gen_exit(jitstate_t *jit, ctx_t *ctx, codeblock_t *cb)
     return code_ptr;
 }
 
+// Generate an interpreter to REG_CFP->pc.
+static uint8_t *
+yjit_gen_context_free_exit(codeblock_t *cb)
+{
+    uint8_t *code_ptr = cb_get_ptr(cb, cb->write_pos);
+
+    // Update the CFP on the EC
+    mov(cb, member_opnd(REG_EC, rb_execution_context_t, cfp), REG_CFP);
+
+    // Put PC into the return register, which the post call bytes dispatches to
+    mov(cb, RAX, member_opnd(REG_CFP, rb_control_frame_t, pc));
+
+    cb_write_post_call_bytes(cb);
+
+    // Note, not incrementing stats here since this exit is the natural end to
+    // executing output code.
+
```

Comment:
Actually I added `GEN_COUNTER_INC(cb, leave_interp_return);` in `leave`. I think it makes sense to keep track of how often we return to the interpreter?